modules/postgres/src/main/scala/doobie/postgres/Instances.scala

```diff
@@ -160,6 +160,13 @@ trait Instances {
   def pgEnumString[A: TypeTag](name: String, f: String => A, g: A => String): Meta[A] =
     enumPartialMeta(name).xmap[A](f, g)
 
+  /**
+   * Construct a `Meta` for values of the given type, mapped via `String` to the named PostgreSQL
+   * enum type with tranparent partiality.
+   */
+  def pgEnumStringOpt[A: TypeTag](name: String, f: String => Option[A], g: A => String): Meta[A] =
```

Comment:
I think `PartialFunction[String, A]` is more convenient than `String => Option[A]`:

```scala
pgEnumStringOpt("my_enum", {
    case "A" => A
    case "B" => B
  }, {
    case A => "A"
    case B => "B"
  })
```